### PR TITLE
Update Vulkan Overview

### PR DESCRIPTION
**Pull request by NixAJ**
I propose the following changes (See Commits) to the Vulkan Overview page.

My main focus was to change the "tone" from more of a "conversation" to an "explanation", and to do some general rewording and restructuring of sentences.

Here is my todo list
- [x] ## What is Vulkan?
- [x] ## When to use Vulkan
- [x] ## Overview of Vulkan API
- [x] ## Multiplatform 
- [x] ## Validation layers 

If you have any feedback on this or would like to discuss my changes please comment.

### CODE REVIEW

File: docs/introduction/vulkan_overview.md [hunk 1]
```diff
@@ -9,58 +9,67 @@ parent: introduction
...
+The Validation Layers are useful for catching a wide variety of errors such as using incorrect configurations, using wrong enums, synchronization issues, and object lifetimes
+and it is recommended to enable Validation Layers while developing to ensure that your code follows the specification requirements. If you can run your application
+without seeing validation errors, you can reasonably assume that your application will work across a wide variety of hardware.
 
-It is important to note that the Validation Layers can't catch every possible bug, like uninitialized memory and bad pointers. There are tools that can check for these bugs, such as Address Sanitizer and valgrind, and using them is highly advised. Unfortunately, graphics drivers often create false positives while using these tools, making their output quite noisy. 
+It is important to note that the Validation Layers does not catch bugs like uninitialized memory and bad pointers. For bugs related to uninitialized memory and bad pointers
```
Comment (cdgiessen) [inline docs/introduction/vulkan_overview.md:60]: This repeats `uninitialized memory and bad pointers` in less than 10 words. It makes the text courser and harder to parse.

File: docs/introduction/vulkan_overview.md [hunk 2]
```diff
@@ -9,58 +9,67 @@ parent: introduction
...
+The API being unified means the two render paths can share code between them.
 
-During this guide we will explain many of the differences between the two types of GPUs.
+This tutorial will be focused on writing Vulkan for Desktop, however we will cover many of the differences between the two types of GPUs.
 
 ## Validation layers
 
-The Vulkan API is very large and rather easy to make mistakes with. Fortunately, when Vulkan was being designed, the creators realized that and made the Validation Layers to fix it. The Validation Layers is a bunch of error checking code that detects and reports 'invalid usage' of the API. They check for all kinds of errors such as using the wrong enums, threading violations, and object lifetimes. The layers are separate from the driver, and to work, they must intercept every function call, perform validation, then call the driver for you. The reason for that is when you want to turn off validation, you Vulkan calls go directly to the driver and can run as fast as possible.
+The vulkan API is vast and as a result it is easy to make mistakes, but this is where Validation Layers comes to the rescue.
```
Comment (cdgiessen) [inline docs/introduction/vulkan_overview.md:50]: I do like the change of the initial description. The "fortunately ..." is clunky. But the change drops *what* the validation layers are, and immediately starts describing how they work (that they intercept functions).

File: docs/introduction/vulkan_overview.md [hunk 3]
```diff
@@ -9,58 +9,67 @@ parent: introduction
...
+code being ran through Vulkan and validate that it is correct and then perform the call to the driver, this also means you will have reduced
+performance when running with Validation Layers enabled.
 
-While developing a Vulkan application, you should have the Validation Layers enabled. The obvious reason is that they detect bugs, but another big reason is that by having a clean validation output you can reasonably assume your application will work across a wide variety of hardware. This is especially important for synchronization, which is tricky to get right, and what may work perfectly fine on one GPU may fail spectacularly on another GPU.
+The Validation Layers are useful for catching a wide variety of errors such as using incorrect configurations, using wrong enums, synchronization issues, and object lifetimes
+and it is recommended to enable Validation Layers while developing to ensure that your code follows the specification requirements. If you can run your application
+without seeing validation errors, you can reasonably assume that your application will work across a wide variety of hardware.
```
Comment (cdgiessen) [inline docs/introduction/vulkan_overview.md:58]: This drops the info about situations which may be drastically different between hardware. Saying that apps should always have a clean output is nice (as most devs want to eliminate all error/warning messages), but a new developer to vulkan wouldn't know what all could be different between hardware.

File: docs/introduction/vulkan_overview.md [hunk 4]
```diff
@@ -9,58 +9,67 @@ parent: introduction
...
 
 ## Validation layers
 
-The Vulkan API is very large and rather easy to make mistakes with. Fortunately, when Vulkan was being designed, the creators realized that and made the Validation Layers to fix it. The Validation Layers is a bunch of error checking code that detects and reports 'invalid usage' of the API. They check for all kinds of errors such as using the wrong enums, threading violations, and object lifetimes. The layers are separate from the driver, and to work, they must intercept every function call, perform validation, then call the driver for you. The reason for that is when you want to turn off validation, you Vulkan calls go directly to the driver and can run as fast as possible.
+The vulkan API is vast and as a result it is easy to make mistakes, but this is where Validation Layers comes to the rescue.
+
+Validation Layers operate as a gateway between your function call and the graphics driver, this allows the Validation Layer to intercept all
+code being ran through Vulkan and validate that it is correct and then perform the call to the driver, this also means you will have reduced
```
Comment (cdgiessen) [inline docs/introduction/vulkan_overview.md:53]: I do like the addition of how performance is impacted when running validation, but it should be noted why its slower. (though, fixing the previous comment I made about gateways would likely fix this one...)

**Review by cdgiessen — CHANGES_REQUESTED** [review]
I hope these comments don't come off as too contradictory, but I felt that there were a lot of small things that could be fixed to improve the clarity and flow of the text.

**Comment by cdgiessen** [comment]
I will add that there were many changes I approve of. Re-reading the original text I found quite a few things that weren't very clean or clear which your changes improved upon.

File: docs/introduction/vulkan_overview.md [hunk 5]
```diff
@@ -9,58 +9,67 @@ parent: introduction
 
 ## What is Vulkan?
 
-Vulkan is a multiplatform API (Application Program Interface) created as a successor to OpenGl by the same authors at the Khronos Group.
-Vulkan is developed by a committee of GPU vendors and other partners, and the API is a standard that anyone can implement.
+Vulkan is a crossplatform API (Application Program Interface), developed by the Khronos Group.
+The Vulkan API is a standard for developers to implement in their preferred environment.
```
[the remark this answers is not in the document]
Comment (NixAJ) [reply docs/introduction/vulkan_overview.md:13]: The previous text said the following "and the API is a standard that anyone can implement", my intent behind my sentence was to improve clarity, any developer can take the Vulkan API and implement it in their preferred environment, where environment is referring to a multitude of factors such as "Device, Platform & Language"

File: docs/introduction/vulkan_overview.md [hunk 6]
```diff
@@ -9,58 +9,67 @@ parent: introduction
...
 
-## When to use Vulkan
-
-While Vulkan is the newest API, you should not be using Vulkan if you don't need its features.
-Writing a Vulkan engine is an order of magnitude harder than using OpenGl, and the GPU-side performance will not necessarily be better.
-
-The big things Vulkan improves is CPU overhead, multithreading, and control over GPU hitches.
-When Vulkan is used right, it is possible to almost completely eliminate hitches, achieving a very smooth framerate, with a very low CPU overhead.
+The Vulkan API provides a list of functions, what arguments to pass in, and the return type.
+The API is based on the C Syntax, and should be compatible with a vast majority of languages.
```
[the remark this answers is not in the document]
Comment (NixAJ) [reply docs/introduction/vulkan_overview.md:20]: [12:39 PM] Nix: I assume the API is actually written in C and not "based on"
[12:39 PM] vblanco: not really written in C

This is my reasoning for what I wrote, I can reword it.

File: docs/introduction/vulkan_overview.md [hunk 7]
```diff
@@ -9,58 +9,67 @@ parent: introduction
...
+This tutorial will be focused on writing Vulkan for Desktop, however we will cover many of the differences between the two types of GPUs.
 
 ## Validation layers
 
-The Vulkan API is very large and rather easy to make mistakes with. Fortunately, when Vulkan was being designed, the creators realized that and made the Validation Layers to fix it. The Validation Layers is a bunch of error checking code that detects and reports 'invalid usage' of the API. They check for all kinds of errors such as using the wrong enums, threading violations, and object lifetimes. The layers are separate from the driver, and to work, they must intercept every function call, perform validation, then call the driver for you. The reason for that is when you want to turn off validation, you Vulkan calls go directly to the driver and can run as fast as possible.
+The vulkan API is vast and as a result it is easy to make mistakes, but this is where Validation Layers comes to the rescue.
+
+Validation Layers operate as a gateway between your function call and the graphics driver, this allows the Validation Layer to intercept all
```
[the remark this answers is not in the document]
Comment (NixAJ) [reply docs/introduction/vulkan_overview.md:52]: I agree, I'll reword it, I wasn't entirely use on the use of "Gateway" initially

**Comment by NixAJ** [comment]
@cdgiessen Thank you so much for your feedback, please take another look through the changes when you have time

**Comment by cdgiessen** [comment]
Only now read the changes after the fact, they look great. Thanks!

**Pull request merged**